### ISSUE TITLE
[bug]: Cannot start backend container beyond `Initialize PubSub` due to `ELIFECYCLE  Command failed.`

### ISSUE THREAD

**Issue by Coniface** [hoppscotch/hoppscotch#3378]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When I try to install hoppscotch according to the documentation, the image `hoppscotch/hoppscotch-backend` systematically crashes during startup. Here are the logs I get:
```log
hoppscotch-backend   | 
hoppscotch-backend   | > hoppscotch-backend@2023.8.1 start:prod /usr/src/app/packages/hoppscotch-backend
hoppscotch-backend   | > node dist/main
hoppscotch-backend   | 
hoppscotch-backend   | Running in production: true
hoppscotch-backend   | Port: 3170
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [NestFactory] Starting Nest application...
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] PubSubModule dependencies initialized +67ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] PrismaModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] MailerModule dependencies initialized +0ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] PassportModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] ThrottlerModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] JwtModule dependencies initialized +0ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] MailerCoreModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] MailerModule dependencies initialized +3ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] AppModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] UserModule dependencies initialized +5ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] UserSettingsModule dependencies initialized +0ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] UserEnvironmentsModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] UserHistoryModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] UserCollectionModule dependencies initialized +0ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] TeamEnvironmentsModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] TeamCollectionModule dependencies initialized +0ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] GraphQLSchemaBuilderModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] UserRequestModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] TeamModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] ShortcodeModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] AuthModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] TeamInvitationModule dependencies initialized +0ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] TeamRequestModule dependencies initialized +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] GraphQLModule dependencies initialized +0ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [InstanceLoader] AdminModule dependencies initialized +1ms
hoppscotch-backend   | Thu, 21 Sep 2023 09:13:02 GMT express-session deprecated undefined resave option; provide resave option at dist/main.js:16:13
hoppscotch-backend   | Thu, 21 Sep 2023 09:13:02 GMT express-session deprecated undefined saveUninitialized option; provide saveUninitialized option at dist/main.js:16:13
hoppscotch-backend   | Enabling CORS with production settings
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RoutesResolver] AppController {/ping}: +11ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/ping, GET} route +3ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RoutesResolver] AuthController {/auth} (version: 1): +0ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/signin, POST} (version: 1) route +2ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/verify, POST} (version: 1) route +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/refresh, GET} (version: 1) route +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/google, GET} (version: 1) route +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/google/callback, GET} (version: 1) route +0ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/github, GET} (version: 1) route +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/github/callback, GET} (version: 1) route +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/microsoft, GET} (version: 1) route +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/microsoft/callback, GET} (version: 1) route +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/logout, GET} (version: 1) route +1ms
hoppscotch-backend   | [Nest] 18  - 09/21/2023, 9:13:02 AM     LOG [RouterExplorer] Mapped {/auth/verify/admin, GET} (version: 1) route +0ms
hoppscotch-backend   | Initialize PubSub
hoppscotch-backend   |  ELIFECYCLE  Command failed.
hoppscotch-backend exited with code 0
```

I already tried:
- to check all issues for a solution, the closest being https://github.com/hoppscotch/hoppscotch/issues/3321 but I don't experience a segfault, only the ELIFECYCLE Command Failed.
- running the AIO image.
- building the image locally and running it.
- building the AIO image locally and running it.

I always get the same issue although the log is slightly different with the AIO image:
```log
hoppscotch  | Backend Server | > hoppscotch-backend@2023.8.1 start:prod /usr/src/app/packages/hoppscotch-backend
hoppscotch  | Backend Server | > node dist/main
hoppscotch  | Backend Server | Running in production: true
hoppscotch  | Backend Server | Port: 3170
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [NestFactory] Starting Nest application...
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] PubSubModule dependencies initialized +77ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] PrismaModule dependencies initialized +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] MailerModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] PassportModule dependencies initialized +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] ThrottlerModule dependencies initialized +2ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] JwtModule dependencies initialized +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] MailerCoreModule dependencies initialized +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] MailerModule dependencies initialized +3ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] AppModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] UserModule dependencies initialized +5ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] UserSettingsModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] UserEnvironmentsModule dependencies initialized +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] UserHistoryModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] UserCollectionModule dependencies initialized +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] TeamEnvironmentsModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] TeamCollectionModule dependencies initialized +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] GraphQLSchemaBuilderModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] UserRequestModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] TeamModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] ShortcodeModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] AuthModule dependencies initialized +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] TeamInvitationModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] TeamRequestModule dependencies initialized +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] GraphQLModule dependencies initialized +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [InstanceLoader] AdminModule dependencies initialized +1ms
hoppscotch  | Backend Server | Thu, 21 Sep 2023 09:05:57 GMT express-session deprecated undefined resave option; provide resave option at dist/main.js:16:13
hoppscotch  | Backend Server | Thu, 21 Sep 2023 09:05:57 GMT express-session deprecated undefined saveUninitialized option; provide saveUninitialized option at dist/main.js:16:13
hoppscotch  | Backend Server | Enabling CORS with production settings
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RoutesResolver] AppController {/ping}: +12ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/ping, GET} route +3ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RoutesResolver] AuthController {/auth} (version: 1): +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/signin, POST} (version: 1) route +2ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/verify, POST} (version: 1) route +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/refresh, GET} (version: 1) route +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/google, GET} (version: 1) route +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/google/callback, GET} (version: 1) route +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/github, GET} (version: 1) route +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/github/callback, GET} (version: 1) route +0ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/microsoft, GET} (version: 1) route +2ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/microsoft/callback, GET} (version: 1) route +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/logout, GET} (version: 1) route +1ms
hoppscotch  | Backend Server | [Nest] 59  - 09/21/2023, 9:05:57 AM     LOG [RouterExplorer] Mapped {/auth/verify/admin, GET} (version: 1) route +0ms
hoppscotch  | Backend Server | Initialize PubSub
hoppscotch  | Exiting process because Backend Server exited with code null
hoppscotch exited with code 0
```

I know the connection to the DB is correct since I managed to run prisma migrations and the schema updated accordingly.

I would expect the service to start with no error.

### Steps to reproduce

1. Follow the installation doc https://docs.hoppscotch.io/documentation/self-host/community-edition/install-and-build
2. Experience the crash upon starting the backend container

### Environment

Production

### Version

Self-hosted

**Comment by balub** [comment]
Hey @Coniface could you provide the following things it could hopefully help in identifying the issue since this is something we cannot really reproduce on our end.
- How much RAM does the host machine have?
- Which version of docker you are using?
In case you are building the containers, can you try it after clearing your cache and deleting the `node_modules` in the `hoppscotch-backend` package and also the `dist` folder in the same package.

**Comment by motienko** [comment]
I have same issue 

Initialize PubSub
 ELIFECYCLE  Command failed.

**Comment by Igorgro** [comment]
@motienko what is v3.0.1?

**Comment by Igorgro** [comment]
Wanted to self-host Hoppscotch instead of Insomnia, but got stuck with this too(

**Comment by bkiro-sportstech** [comment]
I am getting this issue in Linux and Windows. however it works perfectly fine on MacOS (2023 M2: Sonoma)

**Comment by Igorgro** [comment]
It's interesting, how Hoppscotch team runs cloud version, so they don't encounter this issue

**Comment by BeloCraft** [comment]
I catch the same problem using postgresql 14. When I use version 12 or 13 everything is fine.

**Comment by AprilNEA** [comment]
As with this one, I've never been able to get the self-hosted version to work properly

**Comment by pankajmyt** [comment]
Happening with both postgresql 12, 15. 

Docker version `24.0.6, build ed223bc`
Debian 11 8 cpu, 16gb ram

**Comment by pankajmyt** [comment]
What has worked for me is

1. using frontend & admin images from docker hub
2. building backend image from source repo

another issue is emails don't go with AWS SES SMTP creds, so had to update the mailer code to use ses transport.

**Comment by bieblebrox** [comment]
Hi 👋 

Experienced the same bug with Postgres 14 & the backend image from docker cloud.
I can confirm @pankajmyt findings that building hoppscotch-backend image from source (current main) works.

**Comment by perfectra1n** [comment]
Using Postgres 14, and built the AIO container from scratch. It still just crashes :)
![image](https://github.com/hoppscotch/hoppscotch/assets/2390633/91a58447-5d97-4467-9d2c-e4fad78aa200)

```
App/Admin Dashboard Caddy | {"level":"info","ts":1698703775.5149598,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":"caddyfile"}
App/Admin Dashboard Caddy | {"level":"warn","ts":1698703775.5157526,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":2}
App/Admin Dashboard Caddy | {"level":"info","ts":1698703775.5165324,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]}
App/Admin Dashboard Caddy | {"level":"info","ts":1698703775.5167634,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc0003f01c0"}
App/Admin Dashboard Caddy | {"level":"info","ts":1698703775.516916,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/root/.local/share/caddy"}
App/Admin Dashboard Caddy | {"level":"info","ts":1698703775.5169723,"logger":"tls","msg":"finished cleaning storage units"}
App/Admin Dashboard Caddy | {"level":"info","ts":1698703775.5171938,"msg":"autosaved config (load with --resume flag)","file":"/root/.config/caddy/autosave.json"}
App/Admin Dashboard Caddy | {"level":"info","ts":1698703775.5172045,"msg":"serving initial configuration"}
Backend Server | > hoppscotch-backend@2023.8.2 start:prod /usr/src/app/packages/hoppscotch-backend
Backend Server | > node dist/main
Backend Server | Running in production: true
Backend Server | Port: 3170
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[NestFactory] [39m[32mStarting Nest application...[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mPubSubModule dependencies initialized[39m[38;5;3m +197ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mPrismaModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mMailerModule dependencies initialized[39m[38;5;3m +0ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mPassportModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mThrottlerModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mJwtModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mMailerCoreModule dependencies initialized[39m[38;5;3m +0ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mMailerModule dependencies initialized[39m[38;5;3m +4ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mAppModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mUserModule dependencies initialized[39m[38;5;3m +5ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mUserSettingsModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mUserEnvironmentsModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mUserHistoryModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mUserCollectionModule dependencies initialized[39m[38;5;3m +0ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mTeamEnvironmentsModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mTeamCollectionModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mGraphQLSchemaBuilderModule dependencies initialized[39m[38;5;3m +0ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mUserRequestModule dependencies initialized[39m[38;5;3m +2ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mTeamModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mShortcodeModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mAuthModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mTeamInvitationModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mTeamRequestModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mGraphQLModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[InstanceLoader] [39m[32mAdminModule dependencies initialized[39m[38;5;3m +1ms[39m
Backend Server | Mon, 30 Oct 2023 22:09:38 GMT express-session deprecated undefined resave option; provide resave option at dist/main.js:16:13
Backend Server | Mon, 30 Oct 2023 22:09:38 GMT express-session deprecated undefined saveUninitialized option; provide saveUninitialized option at dist/main.js:16:13
Backend Server | Enabling CORS with production settings
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RoutesResolver] [39m[32mAppController {/ping}:[39m[38;5;3m +13ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/ping, GET} route[39m[38;5;3m +3ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RoutesResolver] [39m[32mAuthController {/auth} (version: 1):[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/signin, POST} (version: 1) route[39m[38;5;3m +2ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/verify, POST} (version: 1) route[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/refresh, GET} (version: 1) route[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/google, GET} (version: 1) route[39m[38;5;3m +0ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/google/callback, GET} (version: 1) route[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/github, GET} (version: 1) route[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/github/callback, GET} (version: 1) route[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/microsoft, GET} (version: 1) route[39m[38;5;3m +2ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/microsoft/callback, GET} (version: 1) route[39m[38;5;3m +1ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/logout, GET} (version: 1) route[39m[38;5;3m +0ms[39m
Backend Server | [32m[Nest] 57  - [39m10/30/2023, 10:09:38 PM [32m    LOG[39m [38;5;3m[RouterExplorer] [39m[32mMapped {/auth/verify/admin, GET} (version: 1) route[39m[38;5;3m +1ms[39m
Backend Server | Initialize PubSub
Exiting process because Backend Server exited with code null
```

**Comment by pankajmyt** [comment]
you need to build backend container from source. and use frontend/admin docker images

**Comment by perfectra1n** [comment]
Hi @pankajmyt, what commands did you use to build just the backend container? I only see the production Dockerfile will builds all 3 parts AIO.

**Comment by pankajmyt** [comment]
```
git clone https://github.com/hoppscotch/hoppscotch
cd hoppscotch 
docker build -t myimage -f hoppscotch-backend/Dockerfile .
```

**Comment by perfectra1n** [comment]
Ah, I see that now. That’s annoying, it’s under “ packages/hoppscotch-backend/Dockerfile”. Thanks!

**Comment by ostiwe** [comment]
same, with postgresql 14 not working

**Comment by domints** [comment]
Still same, Postgres 14 as well, image latest (resolved to `2023.8.4` at the moment). 
As @pankajmyt suggested, building backend image straight from git repo works, it doesn't crash anymore, but that's suboptimal solution.

**Comment by glyalin** [comment]
I have managed to connect with Postgres 14 in my dev environment. But i have built the backend image from docker image. The problem is if you use DockerProd.dockerfile
```
FROM node:18-alpine3.16 as base_builder.
```
You should change it for node:18-alpine3.19 and it's connecting with external DB.

Node:18-alpine3.16 connected without problems with dockerized PG instance, but not to external DB...

![image](https://github.com/hoppscotch/hoppscotch/assets/113016306/a5bd4c49-bc0d-4f05-a12f-8f85edac99c8)

**Comment by balub** [comment]
Closing Issue as fix merged in release `2023.12.1`.
